### PR TITLE
HTTPS_SERVER will not be set here as we aren't loading a configure.ph…

### PR DESCRIPTION
**Pull request by zcwilt**
…p in zc_install

### CODE REVIEW

**Review by lat9 — APPROVED** [review]

**Comment by drbyte** [comment]
@lat9 @zcwilt I think this PR incorrectly changed the placement of conditional parentheses.

**Review by drbyte — COMMENTED** [review]
see review notes below

File: zc_install/includes/application_top.php [hunk 1]
```diff
@@ -148,10 +148,10 @@
  */
 $request_type = ((isset($_SERVER['HTTPS']) && (strtolower($_SERVER['HTTPS']) == 'on' || $_SERVER['HTTPS'] == '1')) ||
     (isset($_SERVER['HTTP_X_FORWARDED_BY']) && stripos($_SERVER['HTTP_X_FORWARDED_BY'], 'SSL') !== false) ||
-    (isset($_SERVER['HTTP_X_FORWARDED_HOST']) && (stripos($_SERVER['HTTP_X_FORWARDED_HOST'], 'SSL') !== false || stripos($_SERVER['HTTP_X_FORWARDED_HOST'], str_replace('https://', '', HTTPS_SERVER)) !== false)) ||
+    (isset($_SERVER['HTTP_X_FORWARDED_HOST']) && (stripos($_SERVER['HTTP_X_FORWARDED_HOST'], 'SSL') !== false) ||
```
Comment (drbyte) [inline zc_install/includes/application_top.php:151]: Imbalanced parentheses here. One too many closing-parentheses removed from the end of the line.

The removal should have been the second (green) line shown here:
```diff
- || stripos($_SERVER['HTTP_X_FORWARDED_HOST'], str_replace('https://', '', HTTPS_SERVER)) !== false)
+ || stripos($_SERVER['HTTP_X_FORWARDED_HOST'], str_replace('https://', '', HTTPS_SERVER)) !== false
```

File: zc_install/includes/application_top.php [hunk 2]
```diff
@@ -148,10 +148,10 @@
...
     (isset($_SERVER['SCRIPT_URI']) && stripos($_SERVER['SCRIPT_URI'], 'https:') === 0) ||
     (isset($_SERVER['HTTP_X_FORWARDED_SSL']) && ($_SERVER['HTTP_X_FORWARDED_SSL'] == '1' || strtolower($_SERVER['HTTP_X_FORWARDED_SSL']) == 'on')) ||
-    (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && (strtolower($_SERVER['HTTP_X_FORWARDED_PROTO']) == 'ssl' || strtolower($_SERVER['HTTP_X_FORWARDED_PROTO']) == 'https')) ||
+    (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && (strtolower($_SERVER['HTTP_X_FORWARDED_PROTO']) == 'ssl' || strtolower($_SERVER['HTTP_X_FORWARDED_PROTO']) == 'https'))) ||
```
Comment (drbyte) [inline zc_install/includes/application_top.php:154]: Problem: The added closing parenthesis here is a wrong choice, probably seemed needed due to the mistaken removal of an extra parenthesis on line 151 above.

**Pull request merged**